android/libraries/rib-android/src/main/kotlin/com/uber/rib/core/RibActivity.kt

```diff
@@ -49,37 +52,36 @@ abstract class RibActivity :
   LifecycleScopeProvider<ActivityLifecycleEvent>,
   RxActivityEvents {
   private var router: ViewRouter<*, *>? = null
-  private val lifecycleFlow =
+  private val _lifecycleFlow =
     MutableSharedFlow<ActivityLifecycleEvent>(1, 0, BufferOverflow.DROP_OLDEST)
-  private val callbacksFlow =
+  open val lifecycleFlow: SharedFlow<ActivityLifecycleEvent>
+    get() = _lifecycleFlow
+  private val _callbacksFlow =
     MutableSharedFlow<ActivityCallbackEvent>(0, 1, BufferOverflow.DROP_OLDEST)
-  private val lifecycleObservable = lifecycleFlow.asObservable()
-  private val callbacksObservable = callbacksFlow.asObservable()
+  open val callbacksFlow: SharedFlow<ActivityCallbackEvent>
+    get() = _callbacksFlow
 
   /** @return an observable of this activity's lifecycle events. */
-  override fun lifecycle(): Observable<ActivityLifecycleEvent> = lifecycleObservable
+  final override fun lifecycle(): Observable<ActivityLifecycleEvent> = lifecycleFlow.asObservable()
```

Comment:
How about a lateinit backing var to avoid doing this on every call.  Or we can make a new extension function like asCachedObservable() where we hold onto it?

Reply:
Hmm... Yeah, I think we might have an opportunity of "lazy" here